portal/src/CodeEditor.tsx

```diff
@@ -1,16 +1,95 @@
-import React from "react";
-import ControlledEditor, { EditorProps } from "@monaco-editor/react";
+import React, { useCallback, useRef } from "react";
+import ControlledEditor, {
+  EditorProps,
+  Monaco,
+  OnChange,
+  OnMount,
+} from "@monaco-editor/react";
 
 export interface CodeEditorProps extends EditorProps {
   className?: string;
 }
 
+const REGEX_DETECT_IMPORT =
```

Comment:
Per our offline discussion, please document the approach here we are using here. Like the assumption that the imported module looks like an ambient .d.ts. 

And we should only detect the import of authgear_deno_hook. Ignore other unrecognized modules.